### PR TITLE
(Chore) Fix double onChange call in Select component

### PR DESCRIPTION
**Pull request by janjaap**
This PR fixes an issue where the `onChange` callback handler on the `Select` component was called twice; once on component instantiation and once on actual change of the underlying element's value.

Assertions have been added to validate the changes. 

Note that the `cleanup()` call has been removed; cleanup after each test case is default as well as synchronous (so, when needed to call, needs to wrapped in `act()`). Also, events that alter the component's state, need to be wrapped in `act()`; has also been applied.

### CODE REVIEW

File: packages/asc-ui/src/components/Select/Select.test.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import React from 'react'
```
Comment (jonkoops) [inline packages/asc-ui/src/components/Select/Select.test.tsx:1]: Thanks for cleaning up this code!

File: packages/asc-ui/src/components/Select/Select.tsx [hunk 2]
```diff
@@ -82,16 +80,6 @@ const Select = React.forwardRef<
...
-    useEffect(() => {
-      if (inputValue && ref.current) {
-        setValue(inputValue)
-
-        // The selectedValue label doesn't update properly if you send a new value prop to this component and the component doesn't rerender
-        ref.current.value = inputValue
-        ref.current.dispatchEvent(createEvent('change', { bubbles: true }))
-      }
-    }, [inputValue, ref])
-
```
Comment (jonkoops) [inline packages/asc-ui/src/components/Select/Select.tsx:94]: Now that this code has been removed are we not introducing the original bug it attempted to fix? If possible we should add a test that changes the `value` prop to see if the overlaying label gets updated properly.

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: packages/asc-ui/src/components/Select/Select.tsx (hunk 2 above)
Comment (janjaap) [reply packages/asc-ui/src/components/Select/Select.tsx:94]: I've tested the behaviour in several browsers and it all works as expected. I'll add a test that will cover the display value of the overlaying label.

**Review by jonkoops — APPROVED** [review]

**Comment by jonkoops** [comment]
@janjaap Code looks good, could you add an entry to the changelog?

**Comment by janjaap** [comment]
> @janjaap Code looks good, could you add an entry to the changelog?

Done. And added another test that will verify that programmatically changing the value will actually update the overlay label.

**Pull request merged**